service/internal/proctelemetry/config.go

```diff
@@ -182,5 +191,100 @@ func initPeriodicExporter(_ context.Context, exporter telemetry.MetricExporter,
 		}
 		return sdkmetric.NewPeriodicReader(exp, opts...), nil, nil
 	}
+	if exporter.Otlp != nil {
+		var err error
+		var exp sdkmetric.Exporter
+		switch exporter.Otlp.Protocol {
+		case protocolProtobufHTTP:
+			exp, err = initOTLPHTTPExporter(ctx, exporter.Otlp)
+		case protocolProtobufGRPC:
+			exp, err = initOTLPgRPCExporter(ctx, exporter.Otlp)
+		default:
+			return nil, nil, fmt.Errorf("unsupported protocol %s", exporter.Otlp.Protocol)
+		}
+		if err != nil {
+			return nil, nil, err
+		}
+		return sdkmetric.NewPeriodicReader(exp, opts...), nil, nil
+	}
 	return nil, nil, fmt.Errorf("no valid exporter")
 }
+
+func normalizeEndpoint(endpoint string) string {
+	if !strings.HasPrefix(endpoint, "https://") && !strings.HasPrefix(endpoint, "http://") {
+		return fmt.Sprintf("http://%s", endpoint)
+	}
+	return endpoint
+}
+
+func initOTLPgRPCExporter(ctx context.Context, otlpConfig *telemetry.OtlpMetric) (sdkmetric.Exporter, error) {
+	opts := []otlpmetricgrpc.Option{}
+
+	if len(otlpConfig.Endpoint) > 0 {
+		u, err := url.ParseRequestURI(normalizeEndpoint(otlpConfig.Endpoint))
+		if err != nil {
+			return nil, err
+		}
+		opts = append(opts, otlpmetricgrpc.WithEndpoint(u.Host))
+		if u.Scheme == "http" {
+			opts = append(opts, otlpmetricgrpc.WithInsecure())
+		}
+	}
+
+	if otlpConfig.Compression != nil {
+		opts = append(opts, otlpmetricgrpc.WithCompressor(*otlpConfig.Compression))
+	}
+	if otlpConfig.Timeout != nil {
+		opts = append(opts, otlpmetricgrpc.WithTimeout(time.Millisecond*time.Duration(*otlpConfig.Timeout)))
```

Comment:
Why is `time.Millisecond` needed? Other timeouts just use `time.Duration` https://github.com/open-telemetry/opentelemetry-collector/blob/8949d1da72143abe97fa767e90721d85d79fe6b5/exporter/exporterhelper/common.go#L20 and rely on `ParseDuration`. I think we should do that here if possible

Reply:
The spec defines duration values in milliseconds: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md#duration

If we were to rely on `ParseDuration`, that would cause us to differ from the generated jsonschema, unless I'm misunderstanding your suggestion. jsonschema does support duration in the format of [RFC3339](https://datatracker.ietf.org/doc/html/rfc3339#appendix-A) but i'm not sure if we'd want to support that.

Reply:
I see. There are a few principles here:

1. *Internal Collector configuration consistency*: When specifying the same kind of value on the Collector's configuration (e.g. timeouts), the same syntax should be used everywhere
2. *Consistency with specification*: Inherited from Configuration working group. We should be consistent with whatever way of expressing this is stated in the spec.
3. *Ability to implement with JSON schema*: Whatever we choose should be easy to implement with jsonschema without applying patches on top of the generated code.
4. *Ease of usage*: Specifying common timeout values should be as simple as possible.

We cant' pick all (specifically, I think the `time.Duration` syntax addresses (4) but is incompatible with (2), and (1) is incompatible with (2) unless we make a breaking change somewhere).

I was favoring (1) and (4) over (2) and (3), but I am undecided on whether this is the right choice, specially considering (2) seems very important and making non-Go implementations support `time.Duration`'s syntax seems non-trivial. Do you think we should favor (2) and (3)?

Reply:
In general I agree w/ you that 1 & 4 are preferable. I've pinged the configuration wg w/ a question of what to do here. The spec calls for timeout in millis, but java has an [implementation](https://github.com/open-telemetry/opentelemetry-java/blob/2d6b26b10d47f66290ed76ad7f9e4b5ec22dc161/sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/internal/DefaultConfigProperties.java#L146-L175) that supports for both millis and a duration that's closer to what the collector supports today.

Since this is all behind a feature gate, we could always support millis to begin and switch it if the decision changes in the future, wdyt?

Reply:
Okay, let's go with that :)